test/e2e/azure.go

```diff
@@ -0,0 +1,223 @@
+		return fmt.Errorf("failed to get credentials secret %s, error : %v", secretName.Name, err)
+	}
+
+	a.config = &cluserConfig{
+		TenantID:                    string(secret.Data[TENANT_ID]),
+		SubscriptionID:              string(secret.Data[SUBSCIPTION_ID]),
+		ResourceGroup:               string(secret.Data[RESOURCE_GROUP]),
+		Location:                    string(secret.Data[REGION]),
+		ClientID:                    string(secret.Data[CLIENT_ID]),
+		ClientSecret:                string(secret.Data[CLIENT_SECRET]),
+		UseManagedIdentityExtension: true,
+	}
+
+	var environment azure.Environment
+	if a.config.Cloud == "" {
+		environment = azure.PublicCloud
+	}
+	a.config.Environment = environment
+
+	return nil
+}
+
+func (a *azureTestHelper) prepareZoneClient() error {
+	token, err := getAccessToken(a.config)
+	if err != nil {
+		return err
+	}
+	a.zoneClient = dns.NewZonesClientWithBaseURI(a.config.Environment.ResourceManagerEndpoint, a.config.SubscriptionID)
+	a.zoneClient.Authorizer = autorest.NewBearerAuthorizer(token)
+	return nil
+}
+
+// Credentials should in as json string for azure provider.
+func (a *azureTestHelper) makeCredentialsSecret(namespace string) *corev1.Secret {
+	credData := struct {
+		TenantID       string `json:"tenantId"`
+		SubscriptionID string `json:"subscriptionId"`
+		ResourceGroup  string `json:"resourceGroup"`
+		ClientID       string `json:"aadClientId"`
+		ClientSecret   string `json:"aadClientSecret"`
+	}{
+		TenantID:       a.config.TenantID,
+		SubscriptionID: a.config.SubscriptionID,
+		ResourceGroup:  a.config.ResourceGroup,
+		ClientID:       a.config.ClientID,
+		ClientSecret:   a.config.ClientSecret,
+	}
+	azureCreds, err := json.Marshal(credData)
+	if err != nil {
+
+	}
+
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("azure-config-file-%s", randomString(16)),
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"azure.json": azureCreds,
+		},
+	}
+}
+
+func (a *azureTestHelper) platform() string {
+	return string(configv1.AzurePlatformType)
+}
+
+func (a *azureTestHelper) ensureHostedZone(rootDomain string) (string, []string, error) {
+	location := "global"
+	z, err := a.zoneClient.CreateOrUpdate(context.TODO(), a.config.ResourceGroup, rootDomain,
+		dns.Zone{Location: &location}, "", "")
+	if err != nil {
+		return "", []string{}, err
+	}
+	var zoneID string
+	zoneID = *z.ID
+	nameservers := append(*z.ZoneProperties.NameServers)
+	a.zoneName = rootDomain
+	return zoneID, nameservers, nil
+}
+
+func (a *azureTestHelper) deleteHostedZone(rootDomain string) error {
+	if a.zoneName == "" {
+		fmt.Printf("ZoneName is empty, nothing to be deleted")
+		return nil
+	}
+	if _, err := a.zoneClient.Delete(context.TODO(), a.config.ResourceGroup, a.zoneName, ""); err != nil {
+		return fmt.Errorf("unable to delete zone :%s, failed error: %v", a.zoneName, err)
+	}
+	// verify the zone is present
+	if _, err := a.isZoneNameExists(); err != nil {
```

Comment:
Why should we not trust Azure SDK and verify the deletion?

Reply:
this is additional check kept for one enhancement. 
in case test case fails, it close the test without cleaning, so that time these zones will be left over.
this will not cause any issue since SDK support createOrUpdate().

as mentioned in comment, it will be verified apart of deletion.